File: leetcode-submissions/563_Binary_Tree_Tilt.ts
/**
 * Definition for a binary tree node.
 * class TreeNode {
 *     val: number
 *     left: TreeNode | null
 *     right: TreeNode | null
 *     constructor(val?: number, left?: TreeNode | null, right?: TreeNode | null) {
 *         this.val = (val===undefined ? 0 : val)
 *         this.left = (left===undefined ? null : left)
 *         this.right = (right===undefined ? null : right)
 *     }
 * }
 */
function solve(root:TreeNode|null,res:number[]){
    if(!root) return 0 ;
  let left=   solve(root.left,res);
  let right= solve(root.right,res);
  res[0]+=Math.abs(left-right);
//   if(!root.left && !root.right) return 0;
  return root.val+left+right;
}
function findTilt(root: TreeNode | null): number {
    if(!root) return 0 ;
    const res:number[]=[0];

    solve(root,res);
    return res[0];
};

File: leetcode-submissions/563_Binary_Tree_Tilt.cpp
/**
 * Definition for a binary tree node.
 * struct TreeNode {
 *     int val;
 *     TreeNode *left;
 *     TreeNode *right;
 *     TreeNode() : val(0), left(nullptr), right(nullptr) {}
 *     TreeNode(int x) : val(x), left(nullptr), right(nullptr) {}
 *     TreeNode(int x, TreeNode *left, TreeNode *right) : val(x), left(left), right(right) {}
 * };
 */
class Solution {
public:
    int res=0;
    int helper(TreeNode* root){
        if(not root) return 0;
        int leftSum = helper(root->left);
        int rightSum=helper(root->right);
        res+=abs(rightSum-leftSum);
        return leftSum+rightSum+root->val;

    }
    int findTilt(TreeNode* root) {
        if(not root){
            return res;

        }
        helper(root);
        return res;
    }
};